### PR TITLE
Further Unifying Capable And Capability (includes removal of inefficient storing of capable payload prereqs)

### PR DESCRIPTION
**Pull request by tharmaman**
Covers the following ticket: https://nanopay.atlassian.net/browse/NP-2995

### CODE REVIEW

File: src/foam/u2/crunch/wizardflow/PutFinalPayloadsAgent.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      //   none already exists.
+      // Save no-data capabilities (i.e. not displayed in wizard)
+      return Promise.all(this.wizardlets.filter(wizardlet => {
+        return wizardlet.isAvailable && ( ! wizardlet.capability.of ); 
+      }).map(
+        filteredWizard => {
+          return filteredWizard.save()
+            .then((data) => {
+              this.crunchService.pub('updateJunction');
+              console.log(data);
```
Comment (jlhughes) [inline src/foam/u2/crunch/wizardflow/PutFinalPayloadsAgent.js]: If you need this in immediate future, perhaps console.debug.

Comment (tharmaman) [reply src/foam/u2/crunch/wizardflow/PutFinalPayloadsAgent.js]: Done.

File: src/foam/nanos/crunch/lite/Capable.js [hunk 2]
```diff
@@ -203,7 +215,22 @@ foam.INTERFACE({
       args: [
         {
           name: 'payloads',
-          type: 'CapablePayload[]'
+          type: 'CapabilityJunctionPayload[]'
+        }
+      ]
+    },
+    {
```
Comment (KernelDeimos) [inline src/foam/nanos/crunch/lite/Capable.js:222]: Add method documentation

File: src/foam/nanos/crunch/lite/Capable.js [hunk 3]
```diff
@@ -221,6 +248,21 @@ foam.INTERFACE({
           type: 'String[]'
         }
       ]
+    },
+    {
+      name: 'getDaoKey',
```
Comment (kgrgreer) [inline src/foam/nanos/crunch/lite/Capable.js]: getDAOKey and setDAOKey

File: src/foam/nanos/crunch/lite/ruler/CapableCreateApprovalsRuleAction.js [hunk 4]
```diff
@@ -84,19 +84,20 @@ foam.CLASS({
 
         Capable capableObj = (Capable) clonedObj;
 
+        if (
+          String.valueOf(obj.getProperty("id")) == null ||
+          SafetyUtil.isEmpty(String.valueOf(capableObj.getDaoKey()))
+        ){
+          logger.error("Missing id and/or daoKey from Capable object");
+          throw new RuntimeException("Missing id and/or daoKey from Capable object");
```
Comment (KernelDeimos) [inline src/foam/nanos/crunch/lite/ruler/CapableCreateApprovalsRuleAction.js]: Generate an alarm

File: src/foam/u2/crunch/wizardflow/SkipGrantedAgent.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+  `,
+  imports: [ 'wizardlets' ],
+  exports: [ 'initialPosition' ],
+  requires: [
+    'foam.nanos.crunch.CapabilityJunctionStatus',
+    'foam.nanos.crunch.ui.CapabilityWizardlet',
+    'foam.u2.wizard.WizardPosition'
+  ],
+
+  enums: [
```
Comment (kgrgreer) [inline src/foam/u2/crunch/wizardflow/SkipGrantedAgent.js:22]: I didn't know you could do this, or, if I did, I forgot.

File: src/foam/nanos/crunch/ui/CapableView.js [hunk 6]
```diff
@@ -77,13 +77,13 @@ foam.CLASS({
...
-            this.notify(e.message, '', this.LogLevel.ERROR, true);
-          }
-      }
+      // if ( ! hasPayloads ) {
+      //     try {
+      //       await this.capableObj.setRequirements(this.capableObj.capableRequirements);
+      //     } catch (e) {
+      //       this.notify(e.message, '', this.LogLevel.ERROR, true);
+      //     }
+      // }
```
Comment (KernelDeimos) [inline src/foam/nanos/crunch/ui/CapableView.js]: Delete this code

File: src/foam/u2/crunch/wizardflow/SkipGrantedAgent.js [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+    {
+      name: 'SkipMode',
+      values: ['HIDE','SHOW','SKIP']
```
Comment (KernelDeimos) [inline src/foam/u2/crunch/wizardflow/SkipGrantedAgent.js]: Change values to objects and add documentation properties

**Review by KernelDeimos — APPROVED** [review]

File: src/foam/nanos/crunch/lite/Capable.js (hunk 3 above)
Comment (tharmaman) [reply src/foam/nanos/crunch/lite/Capable.js]: Done.

File: src/foam/nanos/crunch/ui/CapableView.js (hunk 6 above)
Comment (tharmaman) [reply src/foam/nanos/crunch/ui/CapableView.js]: Done.

File: src/foam/u2/crunch/wizardflow/SkipGrantedAgent.js (hunk 7 above)
Comment (tharmaman) [reply src/foam/u2/crunch/wizardflow/SkipGrantedAgent.js]: Done.

File: src/foam/nanos/crunch/lite/Capable.js (hunk 2 above)
Comment (tharmaman) [reply src/foam/nanos/crunch/lite/Capable.js:222]: Done.

**Review by jlhughes — APPROVED** [review]

**Pull request merged**